zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/schema/Query.scala

```diff
@@ -27,29 +28,41 @@ object QueryExeOption {
 }
 
 case class Query(dataset: String,
-                 lookup: Seq[LookupStatement] = Seq.empty,
-                 filter: Seq[FilterStatement] = Seq.empty,
-                 unnest: Seq[UnnestStatement] = Seq.empty,
-                 groups: Option[GroupStatement] = None,
+                 lookups: Seq[LookupStatement] = Seq.empty,
+                 filters: Seq[FilterStatement] = Seq.empty,
+                 unnests: Seq[UnnestStatement] = Seq.empty,
+                 group: Option[GroupStatement] = None,
                  select: Option[SelectStatement] = None,
                  globalAggr: Option[GlobalAggregateStatement] = None,
                  isEstimable: Boolean = false
                 ) extends IReadQuery {
+  var fieldMap: Map[String, Field] = null
```

Comment:
we should rarely use `var`, especially for the query object.

I assume all these fields are used for `QLGenerator`. If that is the case, it's better to make a separate object, e.g., `QueryFields`, along with the immutable `Query` object. The rest of the codebase won't care about these `QueryFields`. 

Even in that case, since they are all `Map`, it's will be better to use `val fieldX: mutable.Map`